TRex2-lib/src/Common/Funs.cc

```diff
@@ -499,6 +500,26 @@ int getLastValidElement(vector<PubPkt*>& column, int columnSize,
   return minValue;
 }
 
+class CompareTS {
+public:
+  bool operator()(const TimeMs& a, PubPkt* b) { return a < b->getTimeStamp(); }
+  bool operator()(PubPkt* a, const TimeMs& b) { return a->getTimeStamp() < b; }
+  bool operator()(const TimeMs& a, const TimeMs& b) { return a < b; }
+  bool operator()(PubPkt* a, PubPkt* b) {
+    return a->getTimeStamp() < b->getTimeStamp();
+  }
+};
+
+vector<PubPkt*>::iterator getBeginPacket(vector<PubPkt*>& column,
+                                         TimeMs minTime) {
+  return std::upper_bound(column.begin(), column.end(), minTime, CompareTS());
```

Comment:
The binary search algorithm is provided by the standard library, no need to reimplement it.